### DISCUSSION TITLE
Which browsers and devices are supported and tested?

### DISCUSSION THREAD

**Discussion by rafgraph** [radix-ui/primitives#1019] [Help] [answered]
Hi, great work on Radix, excited for the primitives to reach a stable release! A couple of questions around browser support and cross browser/device testing.

---

Which browsers and devices do the Radix Primitives support?

---

Which browsers and devices are the Radix Primitives tested on (e.g. click through the stories on each device/browser)?

For example:
- Chrome desktop (Mac, Windows, Windows w/ touch input)
- Firefox desktop (Mac, Windows, Windows w/ touch input)
- Safari desktop (Mac)
- iPhone (Safari)
- iPad (Safari w/ touch input, Safari w/ mouse & keyboard)
- Android Phone (Chrome, Firefox, Samsung Internet)
- Chromebook
- Etc.

It would be helpful to have some sort of quality assurance that the component library has been tested on a variety of common browsers and devices. Thanks!

**Comment by christiankaindl** [comment]
Couldn't find anything in the docs or via Google. Was looking for this, because I noticed that Menu's don't seems to work on Safari 12 on iOS...

**Comment by benoitgrelard** [comment, marked as answer]
Here's our `browserslist` support here for more clarity:

https://github.com/radix-ui/primitives/blob/8f75e5be679b9c269638bddf532a166d80edecb4/babel.config.js#L30

**Comment by rebelliard** [comment]
Hello. Where can I find the current (2024) list of supported browsers? :)

**Comment by 7mioda** [comment]
Hello! Is there an updated browser support list?

**Comment by pascalduez** [comment]
Hey there, 

I was looking for that information as well but couldn't find anything in the available documentation. 

On the `main` branch, looking at the build script, we can find that ESBuild target is configured with `es2022`.  
https://github.com/radix-ui/primitives/blob/36d954d3c1b41c96b1d2e875b93fc9362c8c09e6/internal/builder/builder.js#L48
Which is in line with the base TS config: https://github.com/radix-ui/primitives/blob/36d954d3c1b41c96b1d2e875b93fc9362c8c09e6/internal/typescript-config/base.json#L17

ES2022 translated to browsers gives us the following list:
<img width="1377" height="699" alt="Screen Shot 2025-10-13 at 11 19 18" src="https://github.com/user-attachments/assets/2b48cbd3-3693-4a8a-9984-46fbfcec252f" />

[Caniuse ES2022](https://caniuse.com/?feats=mdn-javascript_builtins_array_at%2Cmdn-javascript_builtins_regexp_hasindices%2Cmdn-javascript_builtins_object_hasown%2Cmdn-javascript_builtins_error_cause%2Cmdn-javascript_operators_await_top_level%2Cmdn-javascript_classes_private_class_fields%2Cmdn-javascript_classes_private_class_methods%2Cmdn-javascript_classes_static_class_fields%2Cmdn-javascript_classes_static_initialization_blocks#summary)
